### PR TITLE
MTDSA-13235- Response Model for Alignment of Retrieve Self Assessment Balance and Transactions

### PR DESCRIPTION
**Pull request by GazalNebHMRC**

### CODE REVIEW

File: test/v2/fixtures/retrieveBalanceAndTransactions/CodingDetailsFixture.scala [hunk 1]
```diff
@@ -43,9 +46,9 @@ object CodingDetailsFixture {
...
     s"""
        |{
-       |   "taxYearReturn": "${codingDetails.taxYearReturn.get}",
+       |   "returnTaxYear": "${codingDetails.returnTaxYear.get}",
        |   "totalLiabilityAmount": ${codingDetails.totalLiabilityAmount.get},
-       |   "taxYearCoding": "${codingDetails.taxYearCoding.get}",
+       |   "codingTaxYear": "${codingDetails.codingTaxYear.get}",
        |   "coded": {
        |      "charge": ${coded.charge.get},
        |      "initiationDate": "${coded.initiationDate.get}"
```
Comment (sean-flynn-HMRC) [inline test/v2/fixtures/retrieveBalanceAndTransactions/CodingDetailsFixture.scala:54]: I wonder if it is a good idea to use ".get" for all the variables in this file. Should we consider replacing this with ".getOrElse" or even using a string directly? https://stackoverflow.com/questions/25016245/what-does-the-get-method-do-in-scala

File: test/v2/fixtures/retrieveBalanceAndTransactions/FinancialDetailsItemFixture.scala [hunk 2]
```diff
@@ -67,7 +67,7 @@ trait FinancialDetailsItemFixture {
   val financialDetailsItemMtdJson: JsValue =
     Json.parse("""
       |{
-      |  "subItem": "subItem",
+      |  "itemId": "subItem",
       |  "dueDate": "2022-02-02",
       |  "amount": 1.23,
       |  "clearingDate": "2021-01-01",
```
Comment (sean-flynn-HMRC) [inline test/v2/fixtures/retrieveBalanceAndTransactions/FinancialDetailsItemFixture.scala:73]: Line 94 - can we assign "subItem" the value "001" to match the spec? According to[ the spec](https://confluence.tools.tax.service.gov.uk/pages/viewpage.action?pageId=367723051) subItem is a string that should not exceed 3 characters

File: test/v2/fixtures/retrieveBalanceAndTransactions/FinancialDetailsFixture.scala [hunk 3]
```diff
@@ -53,21 +53,22 @@ object FinancialDetailsFixture extends FinancialDetailsItemFixture {
        |{
        |      "taxYear": "${taxYear.asDownstream}",
        |      "documentId": "${chargeDetailObject.documentId}",
-       |      "chargeType": "${chargeDetailObject.subTransactionDescription.get}",
-       |      "mainType": "${chargeDetailObject.mainTransactionDescription.get}",
+       |      "chargeType": "${chargeDetailObject.chargeTypeDescription.get}",
```
Comment (sean-flynn-HMRC) [inline test/v2/fixtures/retrieveBalanceAndTransactions/FinancialDetailsFixture.scala:56]: See comment about using ".get" above

File: test/v2/fixtures/retrieveBalanceAndTransactions/FinancialDetailsItemFixture.scala [hunk 4]
```diff
@@ -22,7 +22,7 @@ import v2.models.response.retrieveBalanceAndTransactions.FinancialDetailsItem
 trait FinancialDetailsItemFixture {
```
Comment (sean-flynn-HMRC) [inline test/v2/fixtures/retrieveBalanceAndTransactions/FinancialDetailsItemFixture.scala:22]: Can we also update the FinancialDetailsItem spec? I believe the rules on how to map the values in "statisticalDocument" have been updated in the spec now. With the current spec, an error should be thrown if statisticalDocument is any string that is not "Y" or "N"

File: test/v2/models/response/retrieveBalanceAndTransactions/DocumentDetailsSpec.scala [hunk 5]
```diff
@@ -58,16 +51,16 @@ class DocumentDetailsSpec extends UnitSpec with JsonErrorValidators {
 
       "convert 'i' to true" in {
```
Comment (sean-flynn-HMRC) [inline test/v2/models/response/retrieveBalanceAndTransactions/DocumentDetailsSpec.scala:52]: I think we need to replace 'i' with 'K' here (line 52)

File: test/v2/fixtures/retrieveBalanceAndTransactions/FinancialDetailsItemFixture.scala (hunk 2 above)
Comment (GazalNebHMRC) [reply test/v2/fixtures/retrieveBalanceAndTransactions/FinancialDetailsItemFixture.scala:73]: I have updated it now. Thank you

File: test/v2/fixtures/retrieveBalanceAndTransactions/FinancialDetailsItemFixture.scala (hunk 4 above)
Comment (GazalNebHMRC) [reply test/v2/fixtures/retrieveBalanceAndTransactions/FinancialDetailsItemFixture.scala:22]: Yes, I have updated it now.

File: test/v2/fixtures/retrieveBalanceAndTransactions/CodingDetailsFixture.scala (hunk 1 above)
Comment (GazalNebHMRC) [reply test/v2/fixtures/retrieveBalanceAndTransactions/CodingDetailsFixture.scala]: I have hardcoded the JSONs as per our conversation.Thanks.

File: test/v2/models/response/retrieveBalanceAndTransactions/DocumentDetailsSpec.scala (hunk 5 above)
Comment (GazalNebHMRC) [reply test/v2/models/response/retrieveBalanceAndTransactions/DocumentDetailsSpec.scala]: Updated.

File: test/v2/fixtures/retrieveBalanceAndTransactions/DocumentDetailsFixture.scala [hunk 6]
```diff
@@ -166,7 +166,7 @@ object DocumentDetailsFixture {
        |  "lastClearingReason": "Incoming Payment",
        |  "lastClearedAmount": 3.99,
        |  "statisticalFlag": true,
-       |  "informationCode": "i",
+       |  "informationCode": "K",
```
Comment (sean-flynn-HMRC) [inline test/v2/fixtures/retrieveBalanceAndTransactions/DocumentDetailsFixture.scala:169]: Should "K" be uppercase or lowercase?

Comment (GazalNebHMRC) [reply test/v2/fixtures/retrieveBalanceAndTransactions/DocumentDetailsFixture.scala]: I have added updated mapping of informationCode now.

**Review by sean-flynn-HMRC — APPROVED** [review]
LVGTM

**Pull request merged**